src/host/ledger.h

```diff
@@ -2,194 +2,26 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
-#include "consensus/ledgerenclavetypes.h"
-#include "ds/logger.h"
 #include "ds/messaging.h"
-
-#include <cstdint>
-#include <cstdio>
-#include <errno.h>
-#include <string>
-#include <sys/types.h>
-#include <unistd.h>
-#include <vector>
+#include "ledgerio.h"
 
 namespace asynchost
 {
-  class Ledger
+  class Ledger : public LedgerIO
```

Comment:
What about we call this class `LedgerEntryHandler` and use composition instead of inheritance here?

Reply:
I like using composition instead of inheritance :)